app/code/community/Bolt/Boltpay/Model/FeatureSwitch.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * Bolt magento plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Class Bolt_Boltpay_Model_FeatureSwitch
+ *
+ * The Magento Model class that provides features switches related utility methods
+ *
+ */
+class Bolt_Boltpay_Model_FeatureSwitch extends Bolt_Boltpay_Model_Abstract
+{
+    /**
+     * This method gets feature switches from Bolt and updates the local DB with
+     * the latest values. To be used in upgrade data and webhooks.
+     */
+    public function updateSwitchesFromBolt()
+    {
+        $switchesResponse = $this->boltHelper()->getFeatureSwitches();
+        $switchesData = @$switchesResponse->data->plugin->features;
+
+        if (!is_array($switchesData) || count($switchesData) == 0) {
+            return;
+        }
+
+        $switches = array();
+        foreach ($switchesData as $switch) {
+            $switches[$switch->name] = (object)array(
+                'value' => $switch->value,
+                'defaultValue' => $switch->defaultValue,
+                'rolloutPercentage' => $switch->rolloutPercentage
+            );
+        }
+        Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', json_encode($switches));
```

Comment:
why not `payment/boltpay/featureSwitches/$switchName` that way reading is quicker? si config explosion not good? i cant find any good guidelines

Reply:
As I interpret this approach, the updating to the database and reading into memory of all switches is done once... or at least it is set up for that pattern.  It seems perfectly natural for periodic bulk updates on software upgrade or push request to update.  Regarding reading a particular feature, one single call to the database and then reading from memory seems faster to me than separate calls to the database for each feature.  Memory footprint is probably negligible in this use case.

Reply:
Yes, I agree with Leon. It's better to have only one SQL call. Until we have no more than a few dozens of feature switch this approach should work much faster

Reply:
@roopakv Do you still feel strongly about this? I'd like to get your approval on this PR so it doesn't sit there in an almost ready state waiting for one last comment to be resolved.

Reply:
Nope was a question. I agree with Leon and vitality